### DISCUSSION TITLE
Using multiples IMU

### DISCUSSION THREAD

**Discussion by Fabien-B** [paparazzi/paparazzi#3047] [Q&A] [answered]
Hi,
I am trying to use multiple IMU. How can I specify their relative rotation ?
I assume I should fill the `body_to_sensor` field, but how can I get what values I should put here ?
I already calibrated both IMUs, and merged their calibration in the airframe file:

```
<define name="IMU_ACCEL_CALIB" value="{{.abi_id=24, .calibrated={.neutral=true, .scale=true},.neutral={-72,-70,148}, .scale={{23269,19648,18259},{38058,31995,29742}}}, {.abi_id=9, .calibrated={.neutral=true, .scale=true},.neutral={-15,-10,-98}, .scale={{32610,6121,64910},{6647,1249,13457}}}}"/>
```

I also assume that once I find the correct `body_to_sensor`, I should use the same for the magnetometer calibration?

**Comment by fvantienen** [comment, marked as answer]
You can have a default rotation in your `imu` module, see `modules/imu/imu_cube.c` as an example where all 3 have a different default rotation.

The `IMU_..._CAL` can overwrite this default rotation and this is applied after the calibration of the neutral and scale such that this isn't influenced. In order to set this rotation you need to set `.rotation = true` in the `calibrated` and then set a `body_to_sensor` with a 3x3 RMAT (<<INT32_TRIG_FRAC).
Note that the `body_to_imu` is applied on top of the `body_to_sensor`. This rotation can thus be set per sensor and the `body_to_imu` is applied to all.

**Comment by dewagter** [comment]
NPS issue: with 3 accelerometers the imu does not align in NPS: #3135: @fvantienen any idea why?
